### PR TITLE
Port to Tokio 0.2

### PR DESCRIPTION
**Pull request by vorot93**
Quickly sew this together to get started with Tokio 0.2 runtime. Due to an additional trait bound in http-service I cut Zstd support. Also I removed `App::run`, although I don't think it will really be missed since `async main` has become commonplace.

Tested locally, appears to be working.

### CODE REVIEW

**Comment by Nemo157** [comment]
Opened https://github.com/rustasync/async-compression/issues/27 to fix the zstd issue.

**Comment by vorot93** [comment]
@Nemo157 Added Zstd back!

File: Cargo.toml [hunk 1]
```diff
@@ -25,10 +25,10 @@ cors = ["tide-cors"]
 hyper = ["http-service-hyper"]
 
 [dependencies]
-futures-preview = "0.3.0-alpha.17"
+futures-preview = "0.3.0-alpha.18"
 http = "0.1"
-http-service = "0.3.1"
-http-service-hyper = { version = "0.3.1", optional = true }
+http-service = { git = "https://github.com/rustasync/http-service", branch = "tokio-0.2" }
+http-service-hyper = { git = "https://github.com/rustasync/http-service", branch = "tokio-0.2", optional = true }
```
Comment (yoshuawuyts) [inline Cargo.toml]: Is this still needed? I believe we already published updated versions right?

Comment (vorot93) [reply Cargo.toml]: If you look closely, this actually depends on the branch. Specifically, this one:

rustasync/http-service#43

Comment (secretfader) [reply Cargo.toml]: `0.3.1` was supposed to release earlier, but I only got it done today after seeing https://github.com/rustasync/http-service/issues/45

Is there a reason we aren't using overrides for the Git dependencies in this branch? (Seems that would prevent issues with publishing, but correct me if I'm wrong.)

Comment (yoshuawuyts) [reply Cargo.toml]: Not sure -- Git deps can't be published to crates.io either way.

**Comment by abonander** [comment]
What's the status on this? I wanted to see if my refactor on master of `multipart-async` was suitable for solving #89 yet but I built it against `futures-*-preview 0.3.0-alpha.18` and I don't really want to downgrade (although I can if necessary).

**Comment by secretfader** [comment]
I removed the example that required `runtime` in `dfbaf72` a few moments ago, which should make wrapping up this PR a little bit easier. Feel free to rebase against `master` at your discretion @vorot93. 

If you need help, I'm now available to help push this over the finish line.

**Comment by vorot93** [comment]
@secretfader It looks like the whole repository has been ripped inside out. This PR should be redone from scratch now. Even then I'm not sure that things won't break again in the process.

**Comment by yoshuawuyts** [comment]
@vorot93 hi! I'm currently working on rebasing past commits based on a single repo; this may take a few days. Thanks so much for the work you've done, and apologies for the inconvenience! -- I'm hoping we can take the work that was done here and land it again soonish!

**Comment by abonander** [comment]
Should this be closed now? https://www.reddit.com/r/rust/comments/duvdzz/announcing_asyncstd_10/f79oba0/

**Comment by tyranron** [comment]
@abonander @yoshuawuyts why? Would you be so kind to explain the situation?

**Comment by abonander** [comment]
I'm only somewhat familiar with the circumstances. @yoshuawuyts can explain much better than I can.

**Comment by tyranron** [comment]
@abonander if so, why do you propose to close this PR referring only to a comment where author says he is banned? Is his work negligible or undesirable? From what I see in this PR it was sorta ongoing and OK, and then suddenly closed without any explanation 😕

**Comment by abonander** [comment]
The tone of the author's comment in that thread implies they don't have access to the PR/repo to complete it and/or have no interest in finishing it. Giving any more detail means entering a discussion I have no desire to partake in.

**Comment by tyranron** [comment]
@abonander thanks for making this clear.

**Comment by yoshuawuyts** [comment]
@tyranron the behavior of several individuals in https://github.com/withoutboats/romio/pull/106 but also through other outlets (dms, forums, repos) has been disappointing and below the standards we expect for community participation. This has been the reason why they've lost the privilege of publicly engaging on repositories that are part of this organization.

Regarding upgrading to a modern runtime. That has already been implemented as part of #350, and I'll be kicking off the series of PRs required for a new release over the course of this week.

**Comment by tyranron** [comment]
@yoshuawuyts I've followed through the comments in withoutboats/romio#106 and in relevant Reddit posts, but clearly don't see any bad or disappointing behavior of @vorot93. I'm no one to judge your community standards, but the whole situation with banning contributors feels rather disappointing and even quite awkward.

Anyway, thanks for the clarification. And I'm finishing this offtopic.

**Comment by rubik** [comment]
@yoshuawuyts I'm very late to the discussion, could you clarify if Tokio is/will be supported at all? I cannot find any mention of it in the README. I really like tide and I would like to use it for my projects, but there are some other libraries that only work with Tokio (Elasticsearch). My understanding is that mixing two runtimes is problematic, so it would be great if tide worked with Tokio as well.

**Comment by yoshuawuyts** [comment]
@rubik `async-std` comes with a `tokio02` feature flag that makes Tokio play nice with other runtimes. Using that should make it possible to use Tokio crates with Tide.

**Comment by rubik** [comment]
@yoshuawuyts Ahh I did not know that. Thanks a lot! That's very promising!

**Pull request closed without merging**